src/gluonts/torch/model/estimator.py

```diff
@@ -0,0 +1,181 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+from typing import NamedTuple, Optional, Iterable
+
+import numpy as np
+import pytorch_lightning as pl
+import torch
+import torch.nn as nn
+
+from gluonts.core.component import validated
+from gluonts.dataset.common import Dataset
+from gluonts.itertools import Cached
+from gluonts.model.estimator import Estimator
+from gluonts.torch.model.predictor import PyTorchPredictor
+from gluonts.transform import Transformation, TransformedDataset
+
+
+class TrainOutput(NamedTuple):
+    transformation: Transformation
+    trained_net: nn.Module
+    predictor: PyTorchPredictor
+
+
+class PyTorchLightningEstimator(Estimator):
+    """
+    An `Estimator` type with utilities for creating PyTorch-Lightning-based models.
+
+    To extend this class, one needs to implement three methods:
+    `create_transformation`, `create_training_network`, `create_predictor`,
+    `create_training_data_loader`, and `create_validation_data_loader`.
+    """
+
+    @validated()
+    def __init__(
+        self,
+        trainer: pl.Trainer,
+        lead_time: int = 0,
+        device: torch.device = torch.device("cpu"),
+        dtype: torch.dtype = torch.float32,
```

Comment:
Device and dtype are already handled by `pl.Trainer`

Reply:
Nice! I'll happily get rid of them